plugins/src/main/java/org/smartregister/fhir/gateway/plugins/SyncAccessDecision.java

```diff
@@ -222,6 +251,101 @@ public String postProcess(RequestDetailsReader request, HttpResponse response)
         return resultContent;
     }
 
+    private IBaseResource processRelatedEntityLocationSyncStrategy(
+            RequestDetailsReader request, HttpResponse response) throws IOException {
+        String resultContent;
+        fhirR4Client.getFhirContext().getRestfulClientFactory().setConnectionRequestTimeout(300000);
+        fhirR4Client.getFhirContext().getRestfulClientFactory().setSocketTimeout(300000);
+
+        int subListSize = 100;
```

Comment:
Cool let me move it to the inner Constants class at the bottom. Should be internal to the class and also should not be re-configured so as not to impact the proxy to FHIR server communication _(Based on tests we did with SRE this was the optimal pagination value for server resources deployed for ESM MG Preview)_

We could however make it configurable but with a caveat on how it could impact stability.